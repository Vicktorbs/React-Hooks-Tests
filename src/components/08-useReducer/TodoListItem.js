import React from 'react'
import PropTypes from 'prop-types'

export const TodoListItem = ({todo, index, handleToggle, handleDelete}) => {
    return (
        <>
            <p className={ `${ todo.done && 'complete' }` } onClick={ () => handleToggle(todo.id) }>
                { index + 1 }. { todo.desc }
            </p>
            <button className="btn btn-danger" onClick={ () => { handleDelete(todo.id) } }>
                Borrar
            </button>
        </>
    )
}
TodoListItem.propTypes = {
    todo: PropTypes.object.isRequired, 
    index: PropTypes.number.isRequired, 
    handleToggle: PropTypes.func.isRequired, 
    handleDelete: PropTypes.func.isRequired
}