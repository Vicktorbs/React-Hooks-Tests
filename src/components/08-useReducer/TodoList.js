import React from 'react'
import { TodoListItem } from './TodoListItem'
import PropTypes from 'prop-types'

export const TodoList = ({todos, handleDelete, handleToggle}) => {
    
    return (
        <ul className="list-group list-group-flush">
            {
                todos.map((todo, i) => (
                    <li className="list-group-item" key={ todo.id }>
                        <TodoListItem todo={ todo } index={ i } handleToggle={ handleToggle } handleDelete={ handleDelete } />
                        {/* <p className={ `${ todo.done && 'complete' }` } onClick={ () => handleToggle(todo.id) }>{ i + 1 }. { todo.desc }</p>
                        <button className="btn btn-danger" onClick={ () => { handleDelete(todo.id) } }>
                            Borrar
                        </button> */}
                    </li>
                ))
            }
        </ul>
    )
}
TodoList.propTypes = {
    todos: PropTypes.array.isRequired, 
    handleToggle: PropTypes.func.isRequired, 
    handleDelete: PropTypes.func.isRequired
}