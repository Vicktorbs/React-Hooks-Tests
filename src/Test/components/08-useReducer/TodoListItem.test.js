import { shallow } from "enzyme"
import React from 'react'
import { TodoListItem } from "../../../components/08-useReducer/TodoListItem"
import { demoTodos } from "../../fixtures/demoTodos";

describe('Tests on component <TodoListItem />', () => {
    
    const handleDelete = jest.fn();
    const handleToggle = jest.fn();

    const wrapper = shallow(<TodoListItem 
            todo={ demoTodos[0] }
            index={ 4235 }
            handleDelete={ handleDelete }
            handleToggle={ handleToggle }
        />);

    test('should show it correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should call function handleDelete', () => {
        wrapper.find('button').simulate('click');
        expect(handleDelete).toHaveBeenCalled();
        expect(handleDelete).toHaveBeenCalledWith(demoTodos[0].id);
    })

    test('should call function handleToggle', () => {
        wrapper.find('p').simulate('click');
        expect(handleToggle).toHaveBeenCalledWith(demoTodos[0].id);
    })
    
    test('should show the text correctly', () => {
        const data = wrapper.find('p').text();
        expect(data).toBe('4236. Learn React');
    })
    
    test('should have the class done if TODO.dono = true', () => {
        const todo = demoTodos[0];
        todo.done = true;
        const wrapper = shallow(<TodoListItem 
            todo={ todo }
            index={ 4235 }
            handleDelete={ handleDelete }
            handleToggle={ handleToggle }
        />);
        expect(wrapper.find('p').hasClass('complete')).toBeTruthy();
    })
    
})
