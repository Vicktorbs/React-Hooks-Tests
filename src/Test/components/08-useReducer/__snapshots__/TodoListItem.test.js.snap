// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on component <TodoListItem /> should show it correctly 1`] = `
<Fragment>
  <p
    className="false"
    onClick={[Function]}
  >
    4236
    . 
    Learn React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Borrar
  </button>
</Fragment>
`;
